src/functions/public/Format-TimeSpan.ps1

```diff
@@ -115,27 +115,16 @@
             $null = $orderedUnits.Add($key)
         }
 
-        # If Precision is not specified, calculate it based on non-zero units or all units if IncludeZeroValues
-        if ($PSBoundParameters.ContainsKey('Precision') -eq $false) {
+        # If Precision is not specified, set behavior for auto-precision mode
+        $autoPrecisionMode = $PSBoundParameters.ContainsKey('Precision') -eq $false
+        if ($autoPrecisionMode) {
             if ($IncludeZeroValues) {
                 # Include all units when IncludeZeroValues is specified
                 $Precision = $orderedUnits.Count
             } else {
-                # Calculate how many units have non-zero values
-                $nonZeroUnits = 0
-                $remainder = $originalTicks
-
-                foreach ($unit in $orderedUnits) {
-                    $unitTicks = $script:UnitMap[$unit].Ticks
-                    $value = [math]::Floor($remainder / $unitTicks)
-                    if ($value -gt 0) {
-                        $nonZeroUnits++
-                    }
-                    $remainder = $remainder - ($value * $unitTicks)
-                }
-
-                # Set precision to the number of non-zero units, minimum 1
-                $Precision = [Math]::Max($nonZeroUnits, 1)
+                # For auto-precision without IncludeZeroValues, we'll process all units
+                # but only include non-zero values. Set precision to maximum to process all.
+                $Precision = $orderedUnits.Count
             }
```

Comment:
Do we even need the inner if/else statement? It seems that as long as we dont have a precision parameter, we need to set it from the orderedunits.count.